pkg/machine/hyperv/machine.go

```diff
@@ -199,12 +199,34 @@ func (m *HyperVMachine) readAndSplitIgnition() error {
 
 func (m *HyperVMachine) Init(opts machine.InitOptions) (bool, error) {
 	var (
-		key string
+		key                string
+		cleanCallbackFuncs []func() error
+		err                error
 	)
 
-	if err := m.addNetworkAndReadySocketsToRegistry(); err != nil {
+	// cleanup half-baked files if init fails at any point
+	defer func() {
+		if err == nil {
+			return
+		}
+		for _, clean := range cleanCallbackFuncs {
+			if err := clean(); err != nil {
+				logrus.Error(err)
+			}
+		}
+	}()
+
+	cleanCallbackFuncs = append(cleanCallbackFuncs, m.ImagePath.Delete)
```

Comment:
no todo, but this points out how different hyperv got ...

Reply:
Yeah, it threw me for a bit to find where everything was :')